### ISSUE TITLE
Add a timeout option for logging back in

### ISSUE THREAD

**Issue by ricburton** [balancemymoney/balance-open#37]
I love TouchID but I don’t like touching it every time I tap on Balance.

It’d be nice if there was a way to set a timeout in minutes.

Here’s how 1Password does it:

<img src='https://shipusercontent.com/b730e04fc074be5a2b352db0d1b60a6c/Screen%20Shot%202017-06-16%20at%209.13.19%20PM.png' title='Screen Shot 2017-06-16 at 9.13.19 PM.png' width=762>

**Comment by ricburton** [comment]
I would love to bump this one up to the top. Is this going to be included in the merge @einsteinx2?

**Comment by ricburton** [comment]
@einsteinx2 can this be added as part of the merge of Balance and Balance Open?

**Comment by einsteinx2** [comment]
Not part of the merge specifically, as we already had all the same lock UI in Balance Open. However, we can (and should) definitely add this for 1.0. It won't be much more than a few lines in the lock logic and a small prefs UI change.

**Comment by einsteinx2** [comment]
Sorry I took so long on this. Taking a look now.

**Comment by einsteinx2** [comment]
Got the bulk of it coded, but there are a few edge cases:

1. If we use a timer to lock, the timer will pause if the screen sleeps. The user would expect even if they have lock on sleep off, if the lock delay passes it should lock.

2. The UI code needs to be rearchitected to support the data model locking after a delay in the background

3. The UI code needs to be rearchitected to differentiate between immediate locking and delayed locking

I'm tentatively leaving this in 1.0 for now, but we may need to bump to 1.0.1. Lots of ways to mess this up and it's important that we do it correctly and thoroughly.

**Comment by einsteinx2** [comment]
My changes so far are in the branch enhancement/lock-timeout

**Comment by einsteinx2** [comment]
Taking another look at this now to see if it's doable for 1.0, otherwise I'll bump it to 1.0.1

**Comment by einsteinx2** [comment]
Yeah this is going to have to be bumped to 1.0.1, it's a much deeper change than I initially thought and will need a lot of validation.

**Comment by einsteinx2** [comment]
Moving this into the backlog, we don't need this for 1.0
